tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Session.java

```diff
@@ -432,6 +434,106 @@ public Runner runner() {
     return new Runner();
   }
 
+  /**
+   * Run all graph initializers.
+   *
+   * <p>Initializers must be executed once before running the graph in a training
+   * loop using a session {@link Runner}.</p>
+   *
+   * <p>This method invokes {@link #runInit(String)} using the default name for the init
+   * operation, which is {@link Init#DEFAULT_NAME}. For example:</p>
+   * <pre>{@code
+   * try (Graph g = new Graph()) {
+   *   Ops tf = Ops.create(g);
+   *
+   *   Variable<TInt32> x = tf.variable(tf.constant(10));
+   *   Variable<TInt32> y = tf.variable(tf.constant(20));
+   *   Add<TInt32> z = tf.math.add(x, y);
+   *   tf.init();
+   *
+   *   try (Session s = new Session(g)) {
+   *     s.runInit();
+   *
+   *     try (Tensor<TInt32> t = s.runner().fetch(z).run().get(0).expect(TInt32.DTYPE)) {
+   *       assertEquals(30, t.data().getInt());
+   *     }
+   *   }
+   * }
+   * }</pre>
+   */
+  public void runInit() {
+    runInit(Init.DEFAULT_NAME);
+  }
+
+  /**
+   * Run all graph initializers grouped under the {@code initOpName} operation.
+   *
+   * <p>Initializers must be executed once before running the graph in a training loop using a
+   * session {@link Runner}.</p>
+   *
+   * <p>The {@code initOpName} is the name of a single operation already added to the
+   * graph that executes all graph initializers at once. For example:</p>
+   * <pre>{@code
+   * try (Graph g = new Graph()) {
+   *   Ops tf = Ops.create(g);
+   *
+   *   Variable<TInt32> x = tf.variable(tf.constant(10));
+   *   Variable<TInt32> y = tf.variable(tf.constant(20));
+   *   Add<TInt32> z = tf.math.add(x, y);
+   *   tf.withName("initialize").init();
+   *
+   *   try (Session s = new Session(g)) {
+   *     s.runInit("initialize");
+   *
+   *     try (Tensor<TInt32> t = s.runner().fetch(z).run().get(0).expect(TInt32.DTYPE)) {
+   *       assertEquals(30, t.data().getInt());
+   *     }
+   *   }
+   * }
+   * }</pre>
+   *
+   * @param initOpName name of the initializer operation.
+   */
+  public void runInit(String initOpName) {
+    Operation operation = graph.operation(initOpName);
+    if (operation == null) {
```

Comment:
Can we check the type of the operation too? If the user creates something that doesn't use the init mechanism but does have the name it might be nice to have some kind of warning.

Reply:
That said, it might be an interop issue with graphs created in other TF languages if it warns every time you run a graph created in Python.

Reply:
@dhruvrajan suggestion is to relax a bit the constraint here and to rename `runInit` to `run`, which accepts any type of operation. I'm ok with this too, what do you think?

Reply:
Fine by me. It's essentially just addTarget(String).run() then though?

Reply:
a short cut to `session.runner().addTarget(name).run()`, yes.

Reply:
Given this exists, should we add a convenience method for `session.runner().fetch(name).run()`? Also the exception still mentions an initializer operation, but this isn't required anymore.

Reply:
I guess we could. But fetching is more tricky because you need to make sure to release the returned tensor, a behaviour I would like us to review as a whole at some point, wdyt?

If you are ok with this, then I'll update the exception message (thanks for catching this) and merge this PR now.